### PR TITLE
PDCT-310 Get, search, and count family events

### PR DESCRIPTION
**Pull request by katybaulch**
# Description

Please include:
- Added event router
- Added get, all, search, and count service and repo for family events
- Added family events get, search, all endpoints to events router
- Unit and integration tests for the above

[Link to Linear ticket](https://linear.app/climate-policy-radar/issue/PDCT-310/endpoint-for-getsearch-events)

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## How Has This Been Tested?

Unit and integration tests included.

## Reviewer Checklist

- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: integration_tests/setup_db.py [hunk 1]
```diff
@@ -303,3 +339,26 @@ def _setup_document_data(
...
+
+    for index in range(EXPECTED_NUM_EVENTS):
+        data = EXPECTED_EVENTS[index]
+        fe = FamilyEvent(
+            import_id=data["import_id"],
+            title=data["event_title"],
+            date=data["date"],
+            event_type_name=data["event_type_value"],
+            family_import_id=data["family_import_id"],
+            # family_document_import_id=data["family_document_import_id"],
```
Comment (diversemix) [inline integration_tests/setup_db.py:360]: So we don't test events associated with a FamilyDocument? Might be worth a TODO or a NOTE here?

File: integration_tests/analytics/test_get.py [hunk 2]
```diff
@@ -44,7 +44,6 @@ def test_get_analytics_summary(client: TestClient, test_db: Session, user_header
     assert type(data) is dict
     assert list(data.keys()) == EXPECTED_ANALYTICS_SUMMARY_KEYS
```
Comment (diversemix) [inline integration_tests/analytics/test_get.py:46]: For the integration tests might be worth checking the counts are correct given the data we have setup, no?

**Review by diversemix — APPROVED** [review]
:gem: Wonderful - thanks so much, great quality and really quick too :star2:

File: integration_tests/setup_db.py (hunk 1 above)
Comment (katybaulch) [reply integration_tests/setup_db.py:360]: Noted!

File: integration_tests/analytics/test_get.py (hunk 2 above)
Comment (katybaulch) [reply integration_tests/analytics/test_get.py:46]: Hmm, I hoped lines 47 through 49 captured this :thinking:

**Pull request merged**